monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -0,0 +1,251 @@
+
+        for d in datalist["testing"]:
+            self.infer_files.append({"image": os.path.join(dataroot, d["image"])})
+
+    @abstractmethod
+    def rank_algos(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def predict(self):
+        """
+        predict results after the models are ranked/weighted
+        """
+        raise NotImplementedError
+
+
+class BundleEnsembleBestN(BundleEnsemble):
+    """
+    Ensemble method that select N model out of all using the models' best_metric scores
+
+    Args:
+        n_best: number of models to pick for ensemble (N).
+    """
+
+    def __init__(self, n_best: int = 5):
+
+        super().__init__()
+        self.n_best = n_best
+
+    def sort_score(self):
+        """
+        Sort the best_metrics
+        """
+        scores = concat_val_to_np(self.algos, [BundleEnsembleKeys.SCORE])
+        return np.argsort(scores).tolist()
+
+    def rank_algos(self):
+        """
+        Rank the algos by finding the top N (n_best) validation scores.
+        """
+        ranks = self.sort_score()
+        if len(ranks) < self.n_best:
+            raise ValueError("Number of available algos is less than user-defined N")
+
+        # get the indices that the rank is larger than N
+        indices = [i for (i, r) in enumerate(ranks) if r >= self.n_best]
+
+        # remove the found indices
+        indices = sorted(indices, reverse=True)
+
+        self.algo_ensemble = deepcopy(self.algos)
+        for idx in indices:
+            if idx < len(self.algo_ensemble):
+                self.algo_ensemble.pop(idx)
+
+    def predict(self, pred_param: Optional[Dict[str, Any]] = None):
+        """
+        Use the ensembled model to predict result
+
+        Args:
+            pred_param: prediction parameter dictionary. The key has two groups. The first group only has 'files_slices' key with
+                a value type of `slice`. The files_slices will slice the infer_files and only make prediction on the
+                infer_files[file_slices]. The second group of params will be passed to the `InferClass` to override the parameters
+                of the class functions.
+
+        Returns:
+            A tensor.
+        """
+        if pred_param is None:
+            param = {}
+        else:
+            param = deepcopy(pred_param)
+
+        files = self.infer_files
+        if "files_slices" in param:
+            slices = param.pop("files_slices")
+            files = self.infer_files[slices]
+
+        outputs = []
+        for i in range(len(files)):
+            print(i)
+            preds = []
+            infer_filename = self.infer_files[i]
+            for algo in self.algo_ensemble:
+                infer_instance = algo[BundleEnsembleKeys.ALGO]
+                param.update({"files": [infer_filename]})
+                pred = infer_instance.predict(param)
+                preds.append(pred[0])
+            outputs.append(sum(preds) / len(preds))
```

Comment:
we can design additional class method for aggregation of all probability maps. now the default way is averaging. we should be able to let user change aggregation methods for their own purpose. and we can even add a tutorial to demonstrate that capacity.